core/src/apps/cardano/layout.py

```diff
@@ -181,9 +204,113 @@ async def confirm_certificate(
     await require_confirm(ctx, Paginated(pages))
 
 
+async def confirm_stake_pool_parameters(
+    ctx: wire.Context,
+    pool_parameters: CardanoPoolParametersType,
+    network_id: int,
+    protocol_magic: int,
+) -> None:
+    page1 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    page1.bold("Stake pool registration")
+    page1.normal("Pool id:")
+    page1.bold(hexlify(pool_parameters.pool_id).decode())
+
+    page2 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    page2.normal("Pool reward account:")
+    page2.bold(pool_parameters.reward_account)
+
+    page3 = Text("Confirm transaction", ui.ICON_SEND, ui.GREEN)
+    page3.normal("Pledge: " + format_coin_amount(pool_parameters.pledge))
+    page3.normal("Cost: " + format_coin_amount(pool_parameters.cost))
+    margin_percentage = (
+        100.0 * pool_parameters.margin_numerator / pool_parameters.margin_denominator
+    )
+    percentage_formatted = ("%f" % margin_percentage).rstrip("0").rstrip(".")
```

Comment:
`str(margin_percentage).rstrip(".0")` seems to do what you mean better?

Reply:
not sure if really possible but "10.0000", otherwise a valid devimal number representation, would end up being just "1" with `rstrip(".0")` which I don't really feel safe about

Reply:
ah, my bad, i got confused about what the `rstrip` argument does

the right thing to do is `"%g" % margin_percentage`, which can even be inlined below

Reply:
actually, `%g` does additional rounding to the number and even drops the decimal places as the number of digits increases

```
>>> print("Margin: %g%%" % 12.487870)
'Margin: 12.4879%'
>>> print("Margin: %g%%" % 100000.7878788)
Margin: 100001%
```

Whereas 
```
>>> margin_percentage = 12.487870
>>> ("%f" % margin_percentage).rstrip("0").rstrip(".")
>>> 12.48787
```

So I would rather stick to `rstrip("0").rstrip(".")`, also for the sake of consistency of decimal numbers formatting in the codebase as this is also done in Trezor's global format_amount function:

https://github.com/trezor/trezor-firmware/blob/c6e78e525a2b6d34b321891f88f29fd2b0372381/core/src/trezor/strings.py#L9